extensions/file/src/main/java/org/jtaccuino/extension/file/FileUtilities.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2025 JTaccuino Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jtaccuino.extension.file;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class FileUtilities {
+
+    private static byte[] ZIP_MAGIC_NUMBER = {0x50, 0x4B, 0x03, 0x04};
+
+    public static boolean isZipFile(Path path) {
+        if (!Files.exists(path) || !Files.isReadable(path) || path.toFile().length() < ZIP_MAGIC_NUMBER.length) {
+            return false;
+        }
+        try (InputStream is = Files.newInputStream(path)) {
+            byte[] fileHeader = new byte[ZIP_MAGIC_NUMBER.length];
+            if (is.read(fileHeader) != ZIP_MAGIC_NUMBER.length) {
+                return false;
+            }
+            return java.util.Arrays.equals(fileHeader, ZIP_MAGIC_NUMBER);
+        } catch (IOException e) {
+            System.err.println("Error reading file header for ZIP check.");
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public static void deleteDirectoryRecursively(Path path) {
+        try (Stream<Path> stream = Files.walk(path)) {
+            stream.sorted((a, b) -> b.compareTo(a))
+                    .forEach(p -> {
+                        try {
+                            Files.delete(p);
+                        } catch(IOException ioex) {
+                            System.err.println("Failed to walk directory: " + path);
+                            ioex.printStackTrace();                            
+                        }
+                    });
+        } catch (IOException e) {
+            System.err.println("Failed to walk directory: " + path);
```

Comment:
Why don't you use a logger here?
There a logging framework around...

Reply:
Needs solution for getting feedback into the notebook output. TBD